qiskit_experiments/analysis/curve_fitting.py

```diff
@@ -111,12 +111,14 @@ def fit_func(x, *params):
         kwargs["absolute_sigma"] = True
 
     # Run curve fit
-    # TODO: Add error handling so if fitting fails we can return an analysis
-    #       result containing this information
-    # pylint: disable = unbalanced-tuple-unpacking
-    popt, pcov = opt.curve_fit(
-        fit_func, xdata, ydata, sigma=sigma, p0=param_p0, bounds=param_bounds, **kwargs
-    )
+    try:
+        # pylint: disable = unbalanced-tuple-unpacking
+        popt, pcov = opt.curve_fit(
+            fit_func, xdata, ydata, sigma=sigma, p0=param_p0, bounds=param_bounds, **kwargs
+        )
+    except Exception as ex:
+        raise AnalysisError("scipy.optimize.curve_fit failed") from ex
```

Comment:
I prefer to keep original error message. This error message doesn't guide us to fix the issue.

Reply:
Thats the point of `from ex`. It also raises and shows the original error message in the trace stack.

Reply:
Right, but how can we get all trace back? I think we need something like
```
try:
   do something
except Exception as ex:
   analysis_result["success"] = False
   analysis_result["error message"] = str(ex)
```
but this doesn't capture trace stack.

Reply:
This PR is to do that, but instead of catching a every exception it only catches `AnalysisError`, and the error message raised by `AnalysisError should be what you want to get returned in the analysis result. Any other exception should raise normally stopping analysis as we don't want to general code errors to silently get passed through to analysis results.

Now specifically for `curve_fit` it seems you are right that only the last error string gets included even with `from ex` so I can add the scipy error message to the analysis errors string.